kubernetes/kubernetes_service.go

```diff
@@ -135,9 +148,39 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	}
 	serviceDetails.Autoscalers = autoscalersResponse.autoscalers
 
+	podsResponse := <-podsChan
+	if podsResponse.err != nil {
+		return nil, podsResponse.err
+	}
+	serviceDetails.Pods = podsResponse.pods
+
 	return &serviceDetails, nil
 }
 
+// GetServicePods returns the list of pods associated to a given service.
+// Pods belonging to a service are resolved by looking at the "app" label of the pods.
+// Optionally, a version can be specified to only return the pods belonging to the version (else,
+// set the version to an empty string).
+// It returns an error on any problem.
+func (in *IstioClient) GetServicePods(namespace, serviceName, serviceVersion string) (*v1.PodList, error) {
+	var labelSelectors []string
+	if serviceName != "" {
+		labelSelectors = append(labelSelectors, fmt.Sprintf("app=%v", serviceName))
```

Comment:
Please, don't use the hardcode of app and version for building the labels.
As those may change, we use the constants defined in https://github.com/kiali/kiali/blob/master/config/config.go#L112
Also if we are going to resolve from the selector, I wouldn't to it in this way. First I would fetch the service and I would extract the selector defined in the service.

```
apiVersion: v1
kind: Service
metadata:
  name: details
  labels:
    app: details
spec:
  ports:
  - port: 9080
    name: http
  selector:
    app: details
---
```

So, this assume that you should include this operation in the the GetServiceDetails. 

In the Sprint 1 we did this feature iterating from the endpoints, as the selector may or may not fetch all the pods related to a service, but perhaps that could be a corner case.

Reply:
You are not filtering services. You are filtering pods :-).
Officially, the strong relation of a service and its pods it's due the Endpoint, that means that we can have a pod that belongs to the service that for some reason doesn't have the proper labels.

But I don't mean that, probably that's a corner case, what I mean in my comment that instead to use the fixed stings "app=%" and "version=%", the selector should be fetched from the Service.

If not, you should use the config parameters we used for same purpose.

Reply:
@lucasponce Ok, I'll change this to use the config variables.
And yes, I made some changes to include this in GetServiceDetails. Probably, we need two different version of GetServicePods?

At least for the service graph, we'll need to resolve pods from the labels because the graph uses prometheus metrics to enumerate the services, and I understand that Istio builds these metrics based on the labels. In contrast, what I see from the code of GetServiceDetails, is that a "service" is a K8S service and here it makes sense to use the selector of the service to get the pods.

Reply:
> You are not filtering services. You are filtering pods :-).

Yes, sorry, I meant pods :)   Right, so minimally for my use case, which is to detect "dead" services in the graph, will likely need label filtering on the pods.  In the graph today service nodes can accumulate for services that show up in the Prometheus query but no longer exist in k8s.  Typically for old versions of a service that still have time series in Prometheus but have no recent datapoints/traffic.  We have only the service name and version and we need to decide if it still exists.